core/src/main/java/org/apache/iceberg/MetadataColumns.java

```diff
@@ -38,6 +38,11 @@ private MetadataColumns() {
       Integer.MAX_VALUE - 2, "_pos", Types.LongType.get(), "Ordinal position of a row in the source data file");
   public static final NestedField IS_DELETED = NestedField.required(
       Integer.MAX_VALUE - 3, "_deleted", Types.BooleanType.get(), "Whether the row has been deleted");
+  public static final NestedField SPEC = NestedField.required(
+      Integer.MAX_VALUE - 4, "_spec", Types.IntegerType.get(), "Spec ID to which a row belongs to");
+  public static final int PARTITION_COLUMN_ID = Integer.MAX_VALUE - 5;
+  public static final String PARTITION_COLUMN_NAME = "_partition";
+  public static final String PARTITION_COLUMN_DOC = "Partition to which a row belongs to";
```

Comment:
Nit: I see here that this is so that we can have a later call to `Partitioning.partitionType` (and possibly because it's optional).

But it took me a minute to see why this one is different. Possibly might consider making it more clear or adding a comment? Even just `// Partition type is not static so must be handled specially` might be good, though I know sometimes we prefer the less comments route due to comment rot etc. Up to you 🙂

Reply:
Sure, I'll add a comment above.

Reply:
Added.